pkg/provider/provider.go

```diff
@@ -40,11 +40,11 @@ type Resource struct {
 // RetryUntilTrue returns when there is an error or the requested operation returns true.
 func RetryUntilTrue(name string, retryCount int, fn func() (bool, error)) error {
 	for i := 1; i <= retryCount; i++ {
+		time.Sleep(globalRetryTime)
```

Comment:
why did you move it here?

Reply:
for waiting on update, it takes sometime (readiness probe+threshold) for the pod to become unready/ready. so we wait before we check the status. because when updating, the previous state will be `ready` and we need to wait sometime for it to change.